src/librustc/ty/fold.rs

```diff
@@ -565,7 +565,7 @@ impl<'a, 'gcx, 'tcx> TypeFolder<'gcx, 'tcx> for RegionReplacer<'a, 'gcx, 'tcx> {
 pub fn shift_region(region: ty::RegionKind, amount: u32) -> ty::RegionKind {
     match region {
         ty::ReLateBound(debruijn, br) => {
-            ty::ReLateBound(debruijn.shifted_in(amount), br)
+            ty::ReLateBound(debruijn.shifted_in(amount as usize), br)
```

Comment:
hmm, instead of changing all of these to make `amount: usize` — which we could do — perhaps we should just change `shifted_in` to take a `u32` instead